holoviews/plotting/bokeh/chart.py

```diff
@@ -1055,3 +1055,54 @@ def get_data(self, element, ranges, style):
                             'right': mapping.pop('top'), 'height': mapping.pop('width')})
 
         return sanitized_data, mapping, style
+
+class SegmentPlot(ColorbarPlot):
+    """
+    Segments are lines in 2D space where each two each dimensions specify a
+    (x, y) node of the line.
+    """
+    # Deprecated parameters
+
+    color_index = param.ClassSelector(default=None, class_=(basestring, int),
+                                      allow_None=True, doc="""
+        Deprecated in favor of color style mapping, e.g. `color=dim('color')`""")
+
+    size_index = param.ClassSelector(default=None, class_=(basestring, int),
+                                     allow_None=True, doc="""
+        Deprecated in favor of size style mapping, e.g. `size=dim('size')`""")
```

Comment:
Same comment as below, we shouldn't need these on new element plot classes.

Reply:
Agreed - I thought originally there might be compatibility issues with adapting old code to use the new Segments element.

Reply:
Well, I should have tested before I replied - simply removing them does not work. So how about color_index etc. are deprecated at the same point in time?
```
~/coding/github/holoviews-dev/holoviews/plotting/bokeh/element.py in _get_color_data(self, element, ranges, style, name, factors, colors, int_categories)
   1498                         int_categories=False):
   1499         data, mapping = {}, {}
-> 1500         cdim = element.get_dimension(self.color_index)
   1501         color = style.get(name, None)
   1502         if cdim and ((isinstance(color, util.basestring) and color in element) or isinstance(color, dim)):

AttributeError: 'SegmentPlot' object has no attribute 'color_index'
```

Reply:
You can skip the entire call to ``_get_color_data``.

Reply:
Ah of course, I'd missed a line or two there.

Reply:
I'm looking forward to the day we can delete this mess from the other classes so it's nice to see how much neater new plot classes not burdened with this legacy can be.